be/src/vec/exec/format/parquet/vparquet_group_reader.cpp

```diff
@@ -262,8 +258,9 @@ Status RowGroupReader::_do_lazy_read(Block* block, size_t batch_size, size_t* re
             // generated from next batch, so the filter column is removed ahead.
             DCHECK_EQ(block->rows(), 0);
         } else {
-            Block::filter_block(block, _lazy_read_ctx.all_predicate_col_ids, filter_column_id,
-                                origin_column_num);
+            ColumnPtr filter_column = block->get_by_position(filter_column_id).column;
```

Comment:
We should filter the predicate columns by `_filter_ptr` when there're delete rows.

Reply:
Filter by _filter_ptr in the _filter_block function.